libcomposefs/lcfs-writer.c

```diff
@@ -958,8 +958,13 @@ int lcfs_node_add_child(struct lcfs_node_s *parent, struct lcfs_node_s *child,
 		else
 			new_capacity = parent->children_capacity * 2;
 
+#ifdef HAVE_REALLOCARRAY
```

Comment:
So this is better in that it doesn't cause a safety/security regression on systems that have `reallocarray`, but it is imposing a duplicate code cost on every user.

I'd really prefer we follow what other projects have done and define our own copy of `reallocarray` if it's missing.

Reply:
Yeah, and that copy should do the multiply overflow check.